Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -173,6 +176,42 @@ public class PBXProjGenerator {
 
         return proj
     }
+    
+    func generateTargetAttributes() -> [String: Any]? {
+        
+        
+        var targetAttributes: [String: Any] = [:]
+        
+        // look up TEST_TARGET_NAME build setting
+        func testTargetName(_ target: PBXTarget) -> String? {
+            guard let configurationList = target.buildConfigurationList else { return nil }
+            guard let buildConfigurationReferences = self.proj.objects.configurationLists[configurationList]?.buildConfigurations else { return nil }
+            
+            let configs = buildConfigurationReferences
+                .flatMap { ref in self.proj.objects.buildConfigurations[ref] }
+            
+            return configs
+                .flatMap { $0.buildSettings["TEST_TARGET_NAME"] as? String }
```

Comment:
It's interesting that this could change across configs. I wonder how Xcode deals with this. This is probably fine though

Reply:
I was wondering about that too.
Xcode doesnt let you specify this on a per configuration basis.
Meaning it adds the `TEST_TARGET_NAME ` setting to all configurations.
 and then (i assume) generates the target-attributes.